doc/policy/version3_transactions.md

```diff
@@ -0,0 +1,119 @@
+difficult to replace transactions by attaching lots of descendants to them. For example, if Alice
+wants to replace 4 transactions and each one has 25 or more descendants, the replacement will be
+rejected regardless of its fees.
+
+## Version 3 Rules
+
+All existing standardness rules and policies apply to V3. The following set of additional
+rules apply to V3 transactions:
+
+1. A v3 transaction signals replaceability, even if it does not signal BIP125 replaceability. Other
+   conditions apply, see [RBF rules](./mempool-replacements.md) and [Package RBF
+rules][./packages.md#Package-Replace-By-Fee].
+
+2. Any descendant of an unconfirmed V3 transaction must also be V3.
+
+*Rationale*: Combined with Rule 1, this gives us the property of "inherited signaling" when
+descendants of unconfirmed transactions are created. Additionally, checking whether a transaction
+signals replaceability this way does not require mempool traversal, and does not change based on
+what transactions are mined.
+
+*Note*: A V3 transaction can spend outputs from *confirmed* non-V3 transactions.
+
+*Note*: This rule is not enforced during reorgs.
+
+3. A V3 transaction's unconfirmed ancestors must all be V3.
+
+*Rationale*: Ensure the ancestor feerate rule does not underestimate a to-be-replaced V3 mempool
+transaction's incentive compatibility. Imagine the original transaction, A, has a child B and
+co-parent C (i.e. B spends from A and C). C also has another child, D. B is one of the original
+transactions and thus its ancestor feerate must be lower than the package's. However, this may be an
+underestimation because D can bump C without B's help. This is resolved if V3 transactions can only
+have V3 ancestors, as then C cannot have another child.
+
+*Note*: This rule is enforced during reorgs.
+
+4. A V3 transaction cannot have more than 1 unconfirmed descendant.
+
+Also, [CPFP Carve Out](./mempool-limits.md#CPFP-Carve-Out) does not apply to V3 transactions.
+
+*Rationale*: (upper bound) the larger the descendant limit, the more transactions may need to be
+replaced. This is a problematic pinning attack, i.e., a malicious counterparty prevents the
+transaction from being replaced by adding many descendant transactions that aren't fee-bumping.
+See example #1 in ["Rule 3" Pining](#Rule-3-Pinning) section above.
+
+*Rationale*: (lower bound) at least 1 descendant is required to allow CPFP of the presigned
+transaction. The contract protocol can create presigned transactions paying 0 fees and 1 output for
+attaching a CPFP at broadcast time ("anchor output"). Without package RBF, multiple anchor outputs
+would be required to allow each counterparty to fee-bump any presigned transaction. With package
+RBF, since the presigned transactions can replace each other, 1 anchor output is sufficient.
+
+5. A V3 transaction that has an unconfirmed V3 ancestor cannot have a sigop-adjusted virtual size
+   larger than 1000vB.
```

Comment:
It might be interesting to relax these rules to be:
 * A v3 transaction can have a max descendent count of 5 (not 2)
 * A v3 transaction with an ancestor count of two or more can have a maximum (sigop-adjusted) descendant size of 1000vb
 * A v3 transaction can have at most 1 unconfirmed parent (rather than limiting ancestor count to at most 2)

(where ancestor/descendent size/count includes itself)

Reply:
Why 5?